py3status/argparsers.py

```diff
@@ -60,6 +60,13 @@ def parse_cli_args():
     except subprocess.CalledProcessError:
         i3status_path = None
 
+    # get swaymsg or i3-msg
+    with open(os.devnull, "w") as devnull:
+        if subprocess.call(["pgrep", "-x", "sway"], stdout=devnull) == 0:
```

Comment:
1. remember that -x is [not supported by NixOS](https://github.com/ultrabug/py3status/pull/1638#discussion_r247931639)
1. sway and i3 could be running in parallel in different virtual consoles (very usual for people who are trying to migrate to sway), in those cases we need to preserve "i3" as the default. I'd switch to pgrep for "i3" or "i3status" here.

Reply:
I'm not sure how we can do this. `i3` is a false/default/fallback option. If users are migrating, then it should be something temporary. Could try `"i3" in $(wmctrl -m)`, but we don't guarantee users will have this installed... or maybe `"i3" in $(xprop -root)`. 

Additionally, I don't want to run `pgrep` twice on startup... one for sway, one for i3.

Reply:
How about "if pgrep i3 then i3 else sway"? But I really see no issue with running pgrep twice ("if pgrep i3 elseif pgrep sway else raise error"), it happens once per session anyway.

> If users are migrating, then it should be something temporary.

Yeah but people run both i3 and sway exactly because they want at least one WM to stay working, it would be real pity if by installing sway py3status running on i3 suddenly broke :(

Reply:
Done.